tests/core/test_case.py

```diff
@@ -13,7 +13,7 @@
 from tests.core.utils import (ErrorInSetupCase, SuccessCase, FailureCase,
```

Comment:
I would add unit-tests to all of the new functions you added. 
It's probably painstaking work but it will A. affect our coverage, B. save us from future bugs